compiler/ballerina-lang/src/main/java/io/ballerina/projects/PackageResolution.java

```diff
@@ -161,11 +163,16 @@ private boolean getSticky(PackageContext rootPackageContext) {
         if (rootPackageContext.project().kind() == ProjectKind.BUILD_PROJECT) {
             Path buildFilePath = this.rootPackageContext.project().sourceRoot().resolve(TARGET_DIR_NAME)
                     .resolve(BUILD_FILE);
-            if (buildFilePath.toFile().exists()) {
-                BuildJson buildJson = readBuildJson(buildFilePath);
-                if (!buildJson.isExpiredLastUpdateTime()) {
-                    this.autoUpdate = false;
-                    return true;
+            if (buildFilePath.toFile().exists() && buildFilePath.toFile().length() > 0) {
```

Comment:
We could use `Files.exists(path)` it will check if the file has read access too. Also can we check the behavior where build file without permission

Reply:
Changed to `Files.exists()`

I added a test for NoReadPermissions as well and it creates a new file if the file has no read permissions.